gprofiler/log.py

```diff
@@ -0,0 +1,226 @@
+import datetime
+import json
+import logging
+import logging.handlers
+import os
+import re
+import sys
+from logging import Logger, LogRecord
+from typing import TYPE_CHECKING, Any, Dict, List, MutableMapping, Optional, Tuple
+
+from requests import RequestException
+
+from gprofiler.exceptions import APIError
+from gprofiler.state import State, UninitializedStateException, get_state
+
+if TYPE_CHECKING:
+    from gprofiler.client import APIClient
+
+NO_SERVER_LOG_KEY = "no_server_log"
+NO_SERVER_EXTRA_KEY = "no_extra_to_server"
+RUN_ID_KEY = "run_id"
+CYCLE_ID_KEY = "cycle_id"
+LOGGER_NAME_RE = re.compile(r"gprofiler(?:\..+)?")
+
+
+def get_logger_adapter(logger_name: str) -> logging.LoggerAdapter:
+    # Validate the name starts with gprofiler (the root logger name), so logging parent logger propagation will work.
+    assert LOGGER_NAME_RE.match(logger_name) is not None, "logger name must start with 'gprofiler'"
+    logger = logging.getLogger(logger_name)
+    return GProfilerLoggingAdapter(logger)
+
+
+class GProfilerLoggingAdapter(logging.LoggerAdapter):
+    LOGGING_KWARGS = ["exc_info", "extra", "stack_info"]
```

Comment:
Same as https://github.com/Granulate/gprofiler/pull/119/files#r658174335

Reply:
Can't also, because the adapter method that processes the input receiver `kwargs` as dict and not as kwargs binding. 

Check out the method signature:
`def process(self, msg: Any, kwargs: MutableMapping[str, Any]) -> Tuple[Any, MutableMapping[str, Any]]:`